src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php

```diff
@@ -16,10 +16,17 @@ trait HasTimestamps
     /**
      * Update the model's update timestamp.
      *
+     * @param  string|null  $attribute
      * @return bool
      */
-    public function touch()
+    public function touch($attribute = null)
     {
+        if ($attribute) {
```

Comment:
what would it happen if a given model does not use timestamp? - wouldn't the guard `! $this->usesTimestamps()` be necessary for this use case too?

Reply:
That should be covered by Line 30.

Reply:
`usesTimestamps()` only tells you if the Model is using the standard updated_at and created_at, so it should not be checked when $attribute is passed to the touch method.

It is valid to disable the updated/created timestamps (via $timestamps = false) but still use your own custom timestamp column for some other use case, which is when the $attribute argument would be used.